Mathlib/Analysis/Convex/Continuous.lean

```diff
@@ -0,0 +1,220 @@
+continuous.
+-/
+
+open FiniteDimensional Metric Set List Bornology
+open scoped Topology
+
+variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+  {C : Set E} {f : E → ℝ} {x₀ : E} {ε r : ℝ}
+
+lemma ConvexOn.exists_lipschitzOnWith_of_isBounded (hf : ConvexOn ℝ (ball x₀ r) f) (hε : 0 < ε)
+    (hf' : IsBounded (f '' ball x₀ r)) : ∃ K, LipschitzOnWith K f (ball x₀ (r - ε)) := by
+  rw [isBounded_iff_subset_ball 0] at hf'
+  simp only [Set.subset_def, mem_image, mem_ball, dist_zero_right, Real.norm_eq_abs,
+    forall_exists_index, and_imp, forall_apply_eq_imp_iff₂] at hf'
+  obtain ⟨M, hM⟩ := hf'
+  set K := 2 * M / ε with hK
+  have oneside {x y : E} (hx : x ∈ ball x₀ (r - ε)) (hy : y ∈ ball x₀ (r - ε)) :
+      f x - f y ≤ K * ‖x - y‖ := by
+    obtain rfl | hxy := eq_or_ne x y
+    · simp
+    have hx₀r : ball x₀ (r - ε) ⊆ ball x₀ r := ball_subset_ball <| by linarith
+    have hx' : x ∈ ball x₀ r := hx₀r hx
+    have hy' : y ∈ ball x₀ r := hx₀r hy
+    let z := x + (ε / ‖x - y‖) • (x - y)
+    replace hxy : 0 < ‖x - y‖ := by rwa [norm_sub_pos_iff]
+    have hz : z ∈ ball x₀ r := mem_ball_iff_norm.2 <| by
+      calc
+        _ = ‖(x - x₀) + (ε / ‖x - y‖) • (x - y)‖ := by simp only [z, add_sub_right_comm]
+        _ ≤ ‖x - x₀‖ + ‖(ε / ‖x - y‖) • (x - y)‖ := norm_add_le ..
+        _ < r - ε + ε :=
+          add_lt_add_of_lt_of_le (mem_ball_iff_norm.1 hx) <| by
+            simp [norm_smul, abs_of_nonneg, hε.le, hxy.ne']
+        _ = r := by simp
+    let a := ε / (ε + ‖x - y‖)
+    let b := ‖x - y‖ / (ε + ‖x - y‖)
+    have hab : a + b = 1 := by field_simp [a, b]
+    have hxyz : x = a • y + b • z := by
+      calc
+        x = a • x + b • x := by rw [Convex.combo_self hab]
+        _ = a • y + b • z := by simp [z, a, b, smul_smul, hxy.ne', smul_sub]; abel
+    rw [hK, mul_comm, ← mul_div_assoc, le_div_iff₀' hε]
+    calc
+      ε * (f x - f y) ≤ ‖x - y‖ * (f z - f x) := by
+        rw [mul_sub, mul_sub, sub_le_sub_iff, ← add_mul]
+        have h := hf.2 hy' hz (by positivity) (by positivity) hab
+        field_simp [← hxyz, a, b, ← mul_div_right_comm] at h
+        rwa [← le_div_iff₀' (by positivity), add_comm (_ * _)]
+      _ ≤ _ := by
+        rw [sub_eq_add_neg (f _), two_mul]
+        gcongr
+        · exact (le_abs_self _).trans <| (hM _ hz).le
+        · exact (neg_le_abs _).trans <| (hM _ hx').le
+  refine ⟨K.toNNReal, .of_dist_le' fun x hx y hy ↦ ?_⟩
+  simp_rw [dist_eq_norm_sub, Real.norm_eq_abs, abs_sub_le_iff]
+  exact ⟨oneside hx hy, norm_sub_rev x _ ▸ oneside hy hx⟩
+
+lemma ConcaveOn.exists_lipschitzOnWith_of_isBounded (hf : ConcaveOn ℝ (ball x₀ r) f) (hε : 0 < ε)
+    (hf' : IsBounded (f '' ball x₀ r)) : ∃ K, LipschitzOnWith K f (ball x₀ (r - ε)) := by
+  replace hf' : IsBounded ((-f) '' ball x₀ r) := by convert hf'.neg; ext; simp [neg_eq_iff_eq_neg]
+  simpa using hf.neg.exists_lipschitzOnWith_of_isBounded hε hf'
+
+lemma ConvexOn.isBoundedUnder_abs (hf : ConvexOn ℝ C f) {x₀ : E} (hC : C ∈ 𝓝 x₀) :
```

Comment:
Just a remark that this lemma seems to make a case for a `ConvexAt` predicate.

Reply:
I tend to agree 😁 but I have no other use case in mind for now